packages/core/src/layout/SignInPage/types.ts

```diff
@@ -15,12 +15,37 @@
  */
 
 import { ComponentType } from 'react';
-import { SignInPageProps, SignInResult, ApiHolder } from '@backstage/core-api';
+import {
+  SignInPageProps,
+  SignInResult,
+  ApiHolder,
+  ApiRef,
+  OAuthApi,
+  ProfileInfoApi,
+  BackstageIdentityApi,
+  SessionStateApi,
+} from '@backstage/core-api';
 
-export type ProviderComponent = ComponentType<SignInPageProps>;
+export type SignInConfig = {
+  id: string;
```

Comment:
There is one problem I see here. This `id` should be a string, but not `guest` nor `custom`. In Typescript, I can't do a type exclusion, like `Exclude<string, ('guest | 'custom')>`.

Reply:
Yeah it would have to be a runtime check. Tbh I think it makes sense to do a quick check for duplicates anyway.
When you reduce the array of providers into an object, you could throw an error if a provider with the given id already exists in the object. If there are any duplicate providers it's a configuration error, so let's make sure the app breaks as much as possible :grin: